QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift

```diff
@@ -1,70 +1,24 @@
-import UIKit
-
-class ProgressView: UIView {
-
-    private var label: UILabel!
-    private var stackView: UIStackView!
-
-    override init(frame: CGRect) {
-        super.init(frame: frame)
-
-        createViews()
-        styleViews()
-        defineLayoutForViews()
-    }
-
-    required init?(coder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
-    }
-
-    func set(colors: [UIColor]) {
-        stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
-
-        for color in colors {
-            let cell = UIView()
-            cell.backgroundColor = color
-            cell.layer.cornerRadius = 2
-
-            stackView.addArrangedSubview(cell)
+import SwiftUI
+
+struct ProgressView: View {
+
+    let progressText: String
+    let progressData: [ProgressData]
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text(progressText)
+                .font(.heading5)
+                .foregroundColor(.white)
+
+            HStack(spacing: 5) {
+                ForEach(progressData, id: \.id) {data in
```

Comment:
Since ProgressData is Identifiable, I believe you can write `ForEach(progressData) { data in`

P.S. Note the space after {